### DISCUSSION TITLE
Send code to terminal

### DISCUSSION THREAD

**Discussion by JosiahParry** [zed-industries/zed#8674] [Feature Requests]
I'd like to run interpreted code from Zed, for example this python function. Is there any way to contextually send the code in the editor to the terminal? Are there keyboard commands that can do that?

<img width="795" alt="image" src="https://github.com/zed-industries/zed/assets/13534508/2d8a5855-c609-44e7-aa29-3edb4376a268">

**Comment by Moshyfawn** [comment]
I believe there's already a feature request for this #8344.

**Reply by JosiahParry** [reply]
I think that's close. It seems that issues scope is rather broad. But the idea is similar. I have an open repl, and would like to send lines directly to it.

**Reply by danearvidsson** [reply]
This is a bit old now but is it something similar to this implementation that exist in sublime text, that you are looking for? 

https://packagecontrol.io/packages/SendCode

**Reply by sckott** [reply]
I can't speak for @JosiahParry but yeah that's what I want to be available in zed

**Reply by JosiahParry** [reply]
as another R programmer, yes @sckott I agree :)

**Comment by rrcobb** [comment]
Noting also several vim plugins that offer similar functionality: https://github.com/preservim/vimux or https://github.com/tpope/vim-dispatch. 

Send arbitrary text to another pane / dispatch a command.

The task runner is okay for lots of things! However, for many workflows, it's inferior to being able to send text to an existing terminal pane. 

Maybe `"use_new_terminal": false` is supposed to work like this, but is bugged? Here's the inline documentation:

```js
    // Whether to use a new terminal tab or reuse the existing one to spawn the process, defaults to `false`.
    "use_new_terminal": false,
```

**Comment by remilei** [comment]
I just find a solution to have similar feature than SendCode for Sublime Text. I add this to my keymap file
```{json}
{
        "context": "Workspace",
        "bindings": {
            "ctrl-i": "terminal_panel::ToggleFocus",
            "cmd-alt-enter": [
                "workspace::SendKeystrokes",
                "cmd-c ctrl-i cmd-v enter ctrl-i down cmd-left"
            ]
        }
    }
```

The ctr-i keyboard shortcut is personal, but enable to open or switch to terminal. I just change to cmd-alt-enter rather than cmd-enter, as I potentially have conflict with other shortcut, but I do not find which one for now. The only thing is that we need to highlight text we want to execute.

**Reply by mbannert** [reply]
This is smart. Personally, I like 'send selection' anyway.  I be like `uv run python` and then use your idea. Thanks! 
(FWIW, I guess I'll risk the conflict rather than screwing up my muscle memory which is trained to cmd+enter :)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Reply by JosiahParry** [reply]
Please see https://github.com/zed-industries/zed/pull/42467
